### PR TITLE
Moa Hammarbäck - todo 

### PR DESCRIPTION
**Pull request by mhammarback**
Link to my deployed project:
https://moas-todo.netlify.app/

My reflections on how this week's project turned out:
I'm happy with my end result. I've tried to push myself extra this week and i feel good about my project. 

Things I'd like to have clarified or explained in more detail:
Nothing special! One day i feel that Redux makes so much sense and that i understand it and then the next day i have no idea of what i've done, and then back the next day to feel that i know what i am doing! But i think that is a part of the process and i just need more time and more practice!

### CODE REVIEW

File: code/public/index.html [hunk 1]
```diff
@@ -13,7 +16,7 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-    <title>Technigo React App</title>
+    <title>Moas Todo!</title>
```
Comment (sandra-shumanteva-kaevska) [inline code/public/index.html:19]: Great title :D

File: code/src/components/Item.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import { useDispatch } from 'react-redux'
+import  { todos } from 'reducers/todos'
+
+import { CheckBox } from './Checkbox'
+import { TodoContainer, TodoText, Container } from '../styling/ItemStyling'
+import Trash from '../assets/trash-can.png'
```
Comment (sandra-shumanteva-kaevska) [inline code/src/components/Item.js:8]: Great picture!!

File: code/src/components/List.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            <img className="winner" src = {Winner} alt="gold trophy"/> 
+          </EmptyState> 
+        </EmptyStateContainer>    
+        )
+    }
+
+  return (
+    <ListContainer>
+      {items.map((item, index) => (
+        <Item key={index} item={item}></Item>
```
Comment (sandra-shumanteva-kaevska) [inline code/src/components/List.js]: It is better to use ID instead of index

File: code/src/components/Checkbox.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    setChecked(event.target.checked)
+  }
+
+  const Checkbox = ({ className, checked, ...props }) => (
+    <CheckboxContainer className={className}>
+      <HiddenCheckbox checked={checked} {...props} />
+      <StyledCheckbox checked={checked}>
+        <Icon viewBox='0 0 24 24'>
+          <polyline points='20 6 9 17 4 12' />
+        </Icon>
```
Comment (sandra-shumanteva-kaevska) [inline code/src/components/Checkbox.js:20]: Good job 💯

**Review by sandra-shumanteva-kaevska — APPROVED** [review]
Great Job Moa. Your code is easy to read and follow. You have reached the red level!! 👯 . Keep up with the good work.

File: code/src/App.js [hunk 5]
```diff
@@ -1,9 +1,21 @@
...
 
 export const App = () => {
   return (
-    <div>
-      Find me in src/app.js!
-    </div>
+   <Provider store={store}>
+     <Header></Header>
+     <List></List>
+     <AddItem></AddItem>
```
Comment (HIPPIEKICK) [inline code/src/App.js]: Make these self closing: `<Header/>`

File: code/src/components/AddItem.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+    const handleSubmit = (event) => {
+      event.preventDefault()
+      dispatch(todos.actions.addItem({text}))
+      setText('')
+    }
+    return(
+      <Form onSubmit ={handleSubmit}>
+        <label>
+          <TextInput
```
Comment (HIPPIEKICK) [inline code/src/components/AddItem.js:19]: Prevent the user from submitting empty todos!

File: code/src/components/Header.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+import moment from 'moment'
+import swal from 'sweetalert'
+
+import { Container, Background , Text, DateText, SectionOne, SectionTwo, TaskCount, Button } from '../styling/HeaderStyling'
+
+export const Header = () => {
+    const dispatch = useDispatch()
+    
+    const items = useSelector((store) => store.todos.items)
```
Comment (HIPPIEKICK) [inline code/src/components/Header.js:14]: If you only need the length, add that here already :)

File: code/src/components/Item.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+
+export const Item = ({ item }) => {
+  const dispatch = useDispatch()
+
+  const handleCheckboxClick = () => {
+  dispatch(todos.actions.toggleCompleted(item.id))
```
Comment (HIPPIEKICK) [inline code/src/components/Item.js]: indent

File: code/src/components/List.js (hunk 3 above)
Comment (HIPPIEKICK) [reply code/src/components/List.js]: Bug 🐛 Check the first todo. Then remove it. Then the second todo gets the checked styling even though its `complete` is false.

**Review by HIPPIEKICK — COMMENTED** [review]
Fix bug and prevent the user from submitting empty todos, see comments :)

**Comment by HIPPIEKICK** [comment]
Great Moa! ⭐

**Pull request closed without merging**